nixos/modules/services/misc/matrix-conduit.nix

```
@@ -0,0 +1,168 @@
+            };
+            global.trusted_servers = mkOption {
+              type = types.listOf types.str;
+              default = [ "matrix.org" ];
+              description = "Servers trusted with signing server keys.";
+            };
+
+            global.address = mkOption {
+              type = types.str;
+              default = "127.0.0.1";
+              description = "Address to listen on for connections by the reverse proxy/tls terminator.";
+            };
+          };
+        };
+        default = {};
+        description = ''
+            generates the conduit.toml configuration file. Refer to
+            <link xlink:href="https://gitlab.com/famedly/conduit/-/blob/master/conduit-example.toml"/>
+            for details on supported values.
+            Note that database_path should not be set because the services reliance on systemd StateDir.
+        '';
+      };
+    };
+
+
+    config = mkIf cfg.enable {
+      services.conduit.settings = {
+        global.database_path = "/var/lib/matrix-conduit/conduit.db";
+      };
+      assertions = [
+        {
+          assertion = cfg.settings.global.server_name != null;
+          message = "services.conduit.settings.global.server_name must be set.";
+        }
+        {
+          assertion = cfg.settings.global.allow_federation -> cfg.iHaveReadTheFederationWarning;
+          message = ''
+            Upstream documentation contains the following warning:
+            You should not join Matrix rooms without asking the admins first.
+            We do not know whether Conduit is safe for general use yet, so you
+            should assume there is some chance that it breaks rooms permanently
+            for all participating users. We are not aware of such a bug today,
+            but we would like to do more testing.
+            You need to acknowledge this warning by setting
+            services.conduit.iHaveReadTheFederationWarning = true
+          '';
+        }
+      ];
+
+      systemd.services.conduit = {
+        description = "Conduit Matrix Server";
+        documentation = [ "https://gitlab.com/famedly/conduit/" ];
+        wantedBy = [ "multi-user.target" ];
+        environment = {
+          CONDUIT_CONFIG = configFile;
+        };
+        serviceConfig = {
+          DynamicUser = true;
+          LockPersonality = true;
+          MemoryDenyWriteExecute = true;
+          ProtectClock = true;
+          ProtectControlGroups = true;
+          ProtectHostname = true;
+          ProtectKernelLogs = true;
+          ProtectKernelModules = true;
+          ProtectKernelTunables = true;
+          PrivateDevices = true;
+          PrivateMounts = true;
+          PrivateUsers = true;
+          RestrictAddressFamilies = "AF_INET AF_INET6";
+          RestrictNamespaces = true;
+          RestrictRealtime = true;
+          SystemCallArchitectures = "native";
+          SystemCallFilter = "@system-service";
```

```suggestion
          SystemCallFilter = [
            "@system-service"
            "~@privileged"
          ];
```

One allow and one deny list is usually a good starting point. And excluding privileged is usually non-breaking as well.